c2rust-analyze/src/rewrite/ty.rs

```diff
@@ -140,6 +142,7 @@ fn hir_generic_ty_args<'tcx>(ty: &hir::Ty<'tcx>) -> Option<Vec<&'tcx hir::Ty<'tc
             })
             .collect()
     })
+    .unwrap_or_default()
```

Comment:
I think it makes more sense for this function to still return `Option<_>`, because that preserves the distinction between a `TyKind` that *can't* have generic parameters (where it would be an error for us to add lifetime parameters) and one that can but doesn't yet. To effect the semantic change we want, the `=> args` in this function would become ``=> Some(args.unwrap_or_default()) /* convert missing type arguments to an empty `Vec` */``, and `deconstruct_hir_ty` would be reverted to what it was before this PR.